File: src/main/java/com/baulsupp/oksocial/security/ConsoleCallbackHandler.java
package com.baulsupp.oksocial.security;

import java.io.IOException;
import javax.security.auth.callback.Callback;
import javax.security.auth.callback.CallbackHandler;
import javax.security.auth.callback.PasswordCallback;
import javax.security.auth.callback.UnsupportedCallbackException;

public class ConsoleCallbackHandler implements CallbackHandler {
  @Override public void handle(Callback[] callbacks)
      throws IOException, UnsupportedCallbackException {

    for (Callback c : callbacks) {
      if (c instanceof PasswordCallback) {
        PasswordCallback pw = (PasswordCallback) c;
        pw.setPassword(System.console().readPassword(pw.getPrompt()));
      } else {
        throw new UnsupportedCallbackException(c);
      }
    }
  }
}

File: src/main/java/com/baulsupp/oksocial/security/ConsoleCallbackHandler.kt
package com.baulsupp.oksocial.security

import java.io.IOException
import javax.security.auth.callback.Callback
import javax.security.auth.callback.CallbackHandler
import javax.security.auth.callback.PasswordCallback
import javax.security.auth.callback.UnsupportedCallbackException

class ConsoleCallbackHandler : CallbackHandler {
    @Throws(IOException::class, UnsupportedCallbackException::class)
    override fun handle(callbacks: Array<Callback>) {

        for (c in callbacks) {
            if (c is PasswordCallback) {
                c.password = System.console().readPassword(c.prompt)
            } else {
                throw UnsupportedCallbackException(c)
            }
        }
    }
}
